include/realizations/config/formulation.hpp

```diff
@@ -0,0 +1,135 @@
+#ifndef NGEN_REALIZATION_CONFIG_FORMULATION_H
+#define NGEN_REALIZATION_CONFIG_FORMULATION_H
+
+#include <boost/property_tree/ptree.hpp>
+#include <string>
+
+#include "JSONProperty.hpp"
+
+namespace realization{
+  namespace config{
+
+  struct Formulation{
+    std::string type;
+    //Formulation parameters, object as a PropertyMap
+    geojson::PropertyMap parameters;
+    //List of nested formulations (used for multi bmi representations)
+    std::vector<Formulation> nested;
+
+    /**
+     * @brief Construct a new default Formulation object
+     * 
+     * Default objects have an "" type and and empty property map.
+     */
+    Formulation():type(std::string()), parameters(geojson::PropertyMap()){}
+
+    /**
+     * @brief Construct a new Formulation object
+     * 
+     * @param type formulation type represented
+     * @param params formulation parameter mapping
+     */
+    Formulation(std::string& type, geojson::PropertyMap params):type(std::move(type)), parameters(params){}
```

Comment:
If you're going to move `type`, it should either be passed in by value or rvalue reference. (Pass by value is probably good here, since it'll let the caller decide to move or not move)

Reply:
Changed to pass by value + move in ca837b564057fc79922f55924fa472441bd3c1e1